polkadot/runtime/parachains/src/paras/mod.rs

```diff
@@ -2337,6 +2499,21 @@ impl<T: Config> Pallet<T> {
 	) -> Option<PvfCheckActiveVoteState<BlockNumberFor<T>>> {
 		PvfActiveVoteMap::<T>::get(code_hash)
 	}
+
+	/// This function checks whether the given `code.hash()` exists in the `AuthorizedCodeHash` map
+	/// of authorized code hashes for a para. If found, it verifies that the associated code
+	/// matches the provided `code`. If the validation is successful, it returns tuple as the
+	/// authorized `ValidationCodeHash` with `expire_at`.
+	pub(crate) fn validate_code_is_authorized(
+		code: &ValidationCode,
+		para: &ParaId,
+	) -> Result<AuthorizedCodeHashAndExpiry<BlockNumberFor<T>>, Error<T>> {
+		let authorized = AuthorizedCodeHash::<T>::get(para).ok_or(Error::<T>::NothingAuthorized)?;
+		let now = frame_system::Pallet::<T>::block_number();
+		ensure!(authorized.code_hash == code.hash(), Error::<T>::Unauthorized);
+		ensure!(authorized.expire_at > now, Error::<T>::InvalidBlockNumber);
```

Comment:
swap these two to have the cheap check (expiry check) before the expensive check (hashing the code).

```suggestion
		ensure!(authorized.expire_at > now, Error::<T>::InvalidBlockNumber);
		ensure!(authorized.code_hash == code.hash(), Error::<T>::Unauthorized);
```